docs/architecture/adr-007-address-generation-of-chain-link-proof.md

```diff
@@ -0,0 +1,79 @@
+# ADR 006: Address generation of chain link proof
+
+## Changelog
+
+- January 20th, 2022: Initial draft;
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+Currently, Desmos chain link supports `Bech32`, `Base58` and `Hex` address formats. Besides, each of them only couples one address generation algorithms. 
+Since Desmos will support more addresses from other chains, which might apply the supported address format but with the unsupported algorithm, we SHOULD split the address algorithm from `AddressData` instances.
```

Comment:
> which might apply the supported address format but with the unsupported algorithm

I was wondering: do we have such examples?

Reply:
[elrond address](https://docs.elrond.com/technology/glossary/) is a bech32 format, but it encode public key into bech32 directly instead of `ripemd160(sha256(pubKey))` as cosmos ecosystem.